### PR TITLE
Upgraded proto related deps: grpc and protobuf; removed gogo from core.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: go.mod
Comment (bwplotka) [inline go.mod:6]: Nothing imports but still this guy is keeping coming back here ;p

Comment (johanbrandhorst) [reply go.mod:6]: Try `go mod why github.com/gogo/protobuf`

Comment (bwplotka) [reply go.mod]: It says "nothing is using it" ;p

Anyway, found, updating PR

File: grpctesting/testpb/test.pb.go
Comment (bwplotka) [inline grpctesting/testpb/test.pb.go:239]: Current blocker: somehow new genproto that uses just protobuf (not gogo) does not build me grpc ):

Comment (bwplotka) [reply grpctesting/testpb/test.pb.go:239]:
```
OLDPATH=${PATH}

mkdir -p /tmp/protobin/
cp ${PROTOC_GEN_GO_BIN} /tmp/protobin/protoc-gen-go
cp ${PROTOC_GEN_GO_GRPC_BIN} /tmp/protobin/protoc-gen-go-grpc
PATH=${OLDPATH}:/tmp/protobin

DIRS="grpctesting/testpb"
echo "generating protobuf code"
for dir in ${DIRS}; do
	pushd ${dir}
		${PROTOC_BIN} --go-grpc_out=.\
      -I=. \
			*.proto

			${GOIMPORTS_BIN} -w *.pb.go
	popd
done
```
Anything obvious @johanbrandhorst :thinking:

Comment (bwplotka) [reply grpctesting/testpb/test.pb.go:239]:
```
var file_test_proto_depIdxs = []int32{
	0, // 0: grpc_middleware.testpb.TestService.PingEmpty:input_type -> grpc_middleware.testpb.Empty
	1, // 1: grpc_middleware.testpb.TestService.Ping:input_type -> grpc_middleware.testpb.PingRequest
	1, // 2: grpc_middleware.testpb.TestService.PingError:input_type -> grpc_middleware.testpb.PingRequest
	1, // 3: grpc_middleware.testpb.TestService.PingList:input_type -> grpc_middleware.testpb.PingRequest
	1, // 4: grpc_middleware.testpb.TestService.PingStream:input_type -> grpc_middleware.testpb.PingRequest
	2, // 5: grpc_middleware.testpb.TestService.PingEmpty:output_type -> grpc_middleware.testpb.PingResponse
	2, // 6: grpc_middleware.testpb.TestService.Ping:output_type -> grpc_middleware.testpb.PingResponse
	0, // 7: grpc_middleware.testpb.TestService.PingError:output_type -> grpc_middleware.testpb.Empty
	2, // 8: grpc_middleware.testpb.TestService.PingList:output_type -> grpc_middleware.testpb.PingResponse
	2, // 9: grpc_middleware.testpb.TestService.PingStream:output_type -> grpc_middleware.testpb.PingResponse
	5, // [5:10] is the sub-list for method output_type
	0, // [0:5] is the sub-list for method input_type
	0, // [0:0] is the sub-list for extension type_name
	0, // [0:0] is the sub-list for extension extendee
	0, // [0:0] is the sub-list for field type_name
}

func init() { file_test_proto_init() }
``` 

kind of generates something.. but that's it ;p

Comment (johanbrandhorst) [reply grpctesting/testpb/test.pb.go]: protoc-gen-go v1.25.0 doesn't have grpc. You need `protoc-gen-go-grpc` or `1.4.x` or 
https://github.com/grpc/grpc-go/pull/3453

Comment (bwplotka) [reply grpctesting/testpb/test.pb.go:239]: Used `google.golang.org/grpc/cmd/protoc-gen-go-grpc v0.0.0-20200723182653-9106c3fff523`

I actually made a typo and used only go protoc gen.

BTW This was required as well: `${PROTOC_BIN} --go_out=. --go-grpc_out=.\` so somehow those files are generated separatedly ;p

Which is nice I guess (:  Thanks!

**Comment by bwplotka** [comment]
LOL: Next level error messages in gRPC: `*TestPingService does not implement testpb.TestServiceServer (missing testpb.mustEmbedUnimplementedTestServiceServer method)` XD

**Comment by yashrsharma44** [comment]
@googlebot I consent

**Comment by bwplotka** [comment]
Just to remind the goals of this PR:

* Update `grpc` and `protobuf` deps to latest, use the new API


Given the gogo getting... obsolete (is that right?) maybe we can just stop depending on gogo completely.

**Comment by bwplotka** [comment]
Help wanted :+1:

**Comment by johanbrandhorst** [comment]
Gogo is indeed unlikely to recover, though Kubernetes just reached out to ask what's going on: https://twitter.com/dims/status/1334219405069258753.

**Comment by bwplotka** [comment]
Ok, getting back to this! (: 

cc @yashrsharma44 

Let's move this forward. Upgrading both protobuf and grpc to latest

**Comment by bwplotka** [comment]
Should be fine now (: 

PTAL @yashrsharma44 @johanbrandhorst

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:30]: For now, we should log the error made, else this would make the life of users difficult, wondering why the logging middleware isn't working.

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:45]: Same as above.

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:66]: Same as above

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:81]: Ditto

**Comment by bwplotka** [comment]
Makes sense @yashrsharma44, fixed PTAL 🤗

**Comment by bwplotka** [comment]
Rebased, PTAL 🤗

File: interceptors/logging/payload.go
Comment (yashrsharma44) [inline interceptors/logging/payload.go:152]: So we can break this rule only for errors 😛 - https://www.bwplotka.dev/2020/how-thanos-would-program-in-go/#control-structure-prefer-early-returns-and-avoid-else

**Review by yashrsharma44 — APPROVED** [review]
LGTM 🔥

File: interceptors/logging/payload.go
Comment (bwplotka) [reply interceptors/logging/payload.go:152]: No, we should break it only if it makes sense to do so.

I was thinking about this and since it's only oneliner it does not matter if we do with or without else. WDYT?

I think we could update style guide to mention for multi line bodies? 🤔

**Review by bwplotka — COMMENTED** [review]
Thanks @yashrsharma44 

Let's wait for @johanbrandhorst review until tomorrow, then we can merge & iterate if something is wrong!

File: Makefile
Comment (johanbrandhorst) [inline Makefile:49]: We should be able to remove `PROTOC_GEN_GOGOFAST` right?

File: go.mod
Comment (johanbrandhorst) [inline go.mod:14]: Is this comment necessary?

File: scripts/genproto.sh
Comment (johanbrandhorst) [inline scripts/genproto.sh:12]: Ditto

**Review by johanbrandhorst — COMMENTED** [review]
Nice work! Sorry for the delay on the review. I think we should be able to remove `gogofast` now?

File: go.mod
Comment (bwplotka) [reply go.mod:14]: Up to us

 It's extremely confusing to me when URL is different than GH but somehow project is hosted in GH. 😞

Comment (johanbrandhorst) [reply go.mod:14]: I've never seen it before, and if you open that URL in the browser it takes you to`pkg.go.dev` which has a link to the source repository. I'd rather we removed it tbh.

File: Makefile
Comment (bwplotka) [reply Makefile:49]: We removed just I missed couple of variables somehow

**Comment by bwplotka** [comment]
done! @johanbrandhorst

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**
